packages/rum/test/specs/apm-base.spec.js

```diff
@@ -36,17 +36,41 @@ import {
 import { TRANSACTION_END } from '@elastic/apm-rum-core/src/common/constants'
 import { getGlobalConfig } from '../../../../dev-utils/test-config'
 import Promise from 'promise-polyfill'
+import {
+  hidePageSynthetically,
+  setDocumentVisibilityState
+} from '@elastic/apm-rum-core/test'
 
 var enabled = bootstrap()
 const { serviceName, serverUrl } = getGlobalConfig('rum').agentConfig
 
 describe('ApmBase', function () {
   let serviceFactory
   let apmBase
+  let originalAddEventListener
+  let unobserveVisibilitychange = function noop() {}
 
   beforeEach(function () {
     serviceFactory = createServiceFactory()
     apmBase = new ApmBase(serviceFactory, !enabled)
+
+    // Each time apmBase is initialized the visibilitychange event is attached to window
+    // To avoid side effects when testing we should detach it after each execution
+    originalAddEventListener = window.addEventListener
+    window.addEventListener = (type, listener, options) => {
```

Comment:
offtopic curiosity:

In the future if needed this logic might be extended to intercept all the events occurred while testing and detach them after each execution. It could be a global behaviour for all our tests :)